solidity/contracts/zeto_anon_enc.sol

```diff
@@ -107,4 +110,29 @@ contract Zeto_AnonEnc is ZetoBase {
         );
         return true;
     }
+
+    function deposit(
+        uint256 amount,
+        uint256 utxo,
+        Commonlib.Proof calldata proof
+    ) public {
+        _deposit(amount, utxo, proof);
+        uint256[] memory utxos = new uint256[](1);
+        utxos[0] = utxo;
+        _mint(utxos);
+    }
+
+    function withdraw(
+        uint256 amount,
+        uint256[2] memory inputs,
+        uint256 output,
+        Commonlib.Proof calldata proof
+    ) public {
+        _withdraw(amount, inputs, output, proof);
+        processInputsAndOutputs(inputs, [output, 0]);
+    }
+
+    function mint(uint256[] memory utxos) public onlyOwner {
```

Comment:
@jimthematrix might need a bit more elaboration on the design intention here.

Is the design intention that the UTXO pool can have both erc20 anchored token and the normal tokens? However, there isn't any flag introduced to distinguish the two types of tokens in the UTXO pools. 

I think this means, anyone can use the withdraw function to redeem ERC20 tokens on the anchored contract regardless of whether the input UTXOs are created through anchoring (`deposit` function).

Reply:
it's a fair question. the `mint()` method is reserved for the authority that has deployed the token contract as a way to adjust the supply of the currency. since it's a highly privileged operation, per the `onlyOwner` modifier, it's at the authority's discretion. I feel most real world currency systems would want controls like this

Reply:
to fix the erc20 shortage that could be caused due to usage of this mint function, we could bind the invocation of erc20 mint call in this mint function as well. Happy for that to be a follow-on PR

Reply:
thanks for pointing out the scenario where when an authority mints directly to Zeto UTXOs, but forgets to mint the same amount to ERC20, then an imbalance would be introduced that could cause withdraws to fail since the supplies in the Zeto contract would be larger than the ERC20 balance held by the Zeto contract.

beside providing a sample solution that performs the ERC20 mint by the Zeto contract (which would require a more elaborate ERC20 implementation that allows for additional authorities to mint besides the owner), we should also document that when minting Zeto tokens in a Zeto contract that has an anchored ERC20, the same authority should make sure to mint the same amount in ERC20 and give it to the Zeto contract